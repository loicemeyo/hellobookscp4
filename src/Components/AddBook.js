import React, { Component } from 'react';
import axios from 'axios';
import swal from 'sweetalert';
import { browserHistory } from "react-router";
import { Base_url } from "./Navigation"

/**
 * This component enables an admin to add a new book to the library
 */

class AddBook extends Component {
    state = {
        title:'',
        author:'',
        year:'',
        serial:'',
    }
    /**
     * Allow the admin to view this function only when they are logged in.
     * Otherwise, redirect to login
     */
    componentDidMount() {
        const token = localStorage.getItem("access_token");
        if (!token) {
            return browserHistory.push("/login");
        }
    }
    /**
     * This function sets the state to the new field value as entered by the user
     */
    handleChange = event => {
        this.setState({
            [event.target.name]: event.target.value
        });
    }
    /**
     * Post a new book to the server when a user clicks on 'submit'
     * @param {string} e
     * @returns {object} newbook
     */
    handleSubmit = (event) => {
        event.preventDefault();

        const newbook = {
            "title":this.state.title,
            "author":this.state.author,
            "year":this.state.year,
            "serial_number":this.state.serial
        };

        const token = localStorage.getItem("access_token");
        
        const config ={ headers:{"Authorization":"Bearer " + token}}
       
        axios.post(`${Base_url}/api/v2/books/`, newbook, config)
        .then(response => {
            console.log(response);
            console.log(this.state);
            if(response.data.status === 201){
                swal("new book successfully added");
            } else {
                swal(response.data.message);
            }
        }).catch({});
    };   
   render() {
    return (
        <div style={{ padding: '30px', color: '#337ab7' }}>
        <button className="btn btn-primary" onClick={browserHistory.goBack}>Go Back</button>
        <div className = "jumbotron" id="signupPage">
        <form onSubmit={this.handleSubmit}>
        <h2> Add new book </h2>
            <div className = "row">
                <div className ="col-xs-6">
                    <input
                    id="title"
                    className="form-control"
                    name="title"
                    type="text"
                    placeholder="Enter Book Title"
                    required={true}
                    value={this.state.title}
                    onChange={this.handleChange}
                    />
                </div>
            </div>
            <br/>
            <div className = "row">
                <div className ="col-xs-6">
                    <input
                    id="author"
                    className="form-control"
                    name="author"
                    type="text"
                    placeholder="Enter Book Author"
                    required={true}
                    value={this.state.author}
                    onChange={this.handleChange}
                    />
                </div>
            </div>
            <br/>
            <div className = "row">
                <div className ="col-xs-6">
                    <input
                        id="year"
                        className="form-control"
                        name="year"
                        type="number"
                        placeholder="Enter Publication Year"
                        required={true}
                        value={this.state.year}
                        onChange={this.handleChange}
                    />
                </div>
            </div>
            <br/>
            <div className = "row">
                <div className ="col-xs-6">
                    <input
                        id="serial"
                        className="form-control"
                        name="serial"
                        type="text"
                        placeholder="Enter Serial Number"
                        required={false}
                        value={this.state.serial}
                        onChange={this.handleChange}
                    />
                </div>
            </div>
            <br/>
            <button className='btn btn-primary' type="submit">Add Book</button> 
        </form>
    </div>
    </div>

           );
        }
    }

export default AddBook;
