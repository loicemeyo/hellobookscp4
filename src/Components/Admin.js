import React, { Component } from "react";
import { Link, browserHistory } from "react-router";
import swal from "sweetalert";
import axios from "axios";
class Admin extends Component {
  constructor() {
    super();
    this.state = {
      allBooks: []
    };
  }
  componentDidMount() {
    const token = localStorage.getItem("access_token");
    if (!token) {
      return browserHistory.push("/login");
    }

    const config = { headers: { "Authorization": "Bearer " + token } };


    axios.get("http://127.0.0.1:5000/api/v2/books", config)
      .then(response => {
        this.setState({ allBooks: response.data.books });
      }).catch(error => {
        if (error.response !== undefined) {
          if (error.response.status === 401) {
            const message = error.response.data.Error;
            swal("message!!", message, "error");
            localStorage.removeItem("access_token");
            browserHistory.push("/login");
          }
        }

      });
  }
  /**
   * Make a server request to delete a book
   * @param {int} bookId
   * @returns {string} success message
   */
  handleDelete(bookId) {
    const token = localStorage.getItem("access_token");
    const config = { headers: { "Authorization": "Bearer " + token } };
    const delete_book_url = `http://127.0.0.1:5000/api/v2/books/${bookId}`;
    swal({
      title: "Are you sure?",
      text: "Once deleted, you will not be able to recover this book!",
      icon: "warning",
      buttons: true,
      dangerMode: true,
    }).then((willDelete) => {
      if (willDelete) {
        axios.delete(delete_book_url, config)
          .then(response => {
            swal("Success", response.data.message, "success");
            browserHistory.push("/admin");
          })
          .catch(error => {
            if (error.response.status === 404) {
              const message = error.response.data.Message;
              swal("Error!!", message, "error");
            }
          });
      }
    });
  }
  render() {
    return (
      <div>
        <h2 style={{ padding: "5px", color: "#337ab7" }}>Admin Dashboard</h2>
        <div>
          <Link to={"/addbook"}><button className="btn btn-primary" id="button">Add Book</button></Link>
          <Link to={"/upgradeuser"}><button className="btn btn-primary" id="button">Change user status</button></Link>
          <Link to={"/users"}><button className="btn btn-primary" id="button">View users</button></Link>
        </div>
        <br />
        <h2 style={{ padding: "5px", color: "#337ab7", textAlign: "center" }}>Available Books</h2>
        <div className="col=md-10" >
          <table className="table" id="table">
            <thead>
              <tr>
                <th>Book</th>
                <th>Author</th>
                <th>Publication Year</th>
                <th>Edit Book</th>
                <th>Delete Book</th>
              </tr>
            </thead>
            <tbody>
              {this.state.allBooks.map(book =>
                <tr key={book.ID}>
                  <td><Link to={`/books/${book.ID}`}>{book.Title}></Link></td>
                  <td>{book.Author}</td>
                  <td>{book.Publication}</td>
                  <td><Link to={`/editbook/${book.ID}`}><button className="btn btn-default">Edit Book</button></Link></td>
                  <td><button className="btn btn-default" onClick={this.handleDelete.bind(this, book.ID)}>DeleteBook</button></td>

                </tr>
              )}
            </tbody>
          </table>
        </div>
      </div>

    );
  }
}
export default Admin;