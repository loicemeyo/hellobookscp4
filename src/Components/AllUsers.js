import React, { Component } from "react";
import axios from "axios";
import { browserHistory } from "react-router";

/**
 * This component displays all the users of the library
 */
class AllUsers extends Component {
  constructor() {
    super();
    this.state = {
      allUsers: []
    };
  }

  /**
   * Make server request to display all users
   * Allow admin to view this page only when they are logged in
   * @returns {object} allUsers
   */

  componentDidMount() {
    const token = localStorage.getItem("access_token");
    if (!token) {
      return browserHistory.push("/login");
    }

    const config = { headers: { "Authorization": "Bearer " + token } };


    axios.get("http://127.0.0.1:5000/api/v2/auth/register", config)
      .then(response => {
        
        this.setState({
          allUsers: response.data.library_users
        });
      }).catch();
  }
  render() {
    return (
      <div style={{ padding: "20px", color: "#337ab7" }}>
        <h2 style={{ padding: "5px", color: "#337ab7", textAlign: "center" }}>HelloBooks Users</h2>
        <br />
        <div className="col=md-8">
          <table className="table" style={{ border: "1px solid grey" }}>
            <tr><th>Email</th> <th>Name</th> <th>Admin</th></tr>
            {this.state.allUsers.map(user =>
              <tr key={user.id}>
                <td>{user.Email}</td> <td>{user.Name}</td> <td>{user.Status ? "Yes" : "No"}</td>
              </tr>
            )}
          </table>
        </div>
      </div>

    );
  }
}
export default AllUsers;