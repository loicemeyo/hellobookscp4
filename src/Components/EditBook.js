import React, { Component } from 'react';
import axios from 'axios';
import swal from 'sweetalert';
import {browserHistory} from 'react-router';

/**
 * This component enables an admin to edit a book to the library
 */

class EditBook extends Component {
    constructor (props){
    super(props)
    this.state = {
       books:[],
        title:'',
        author:'',
        year:'',
        serial:'',
    }}
     /**
     * Allow the admin to view this function only when they are logged in.
     * Otherwise, redirect to login
     */
    componentDidMount() {
        const token = localStorage.getItem("access_token");
        if (!token) {
            return browserHistory.push("/login");
        }
    }
    /**
     * This function sets the state to the new field value as entered by the user
     * @param {string} e
     * @returns {string} value
     */
    handleChange = event => {
        this.setState({
            [event.target.name]: event.target.value
        });
    }
    /**
     * Make server request to get the book by ID and populate input fields withe the current book details
     * @param {int} bookId
     * @returns {object}book
     */
    componentDidMount() {
        const token = localStorage.getItem('access_token');
        const bookId = this.props.params.id;
        const single_book_url = `http://127.0.0.1:5000/api/v2/books/${bookId}`

        const config = { headers: { 'Authorization': "Bearer " + token } };

        axios.get(single_book_url, config)
            .then(response => {
                console.log(response.data.book)
                this.setState({
                    books:response.data,
                    id:response.data.book.ID,
                    title:response.data.book.Title,
                    author:response.data.book.Author,
                    year:response.data.book.Publication,
                    serial:response.data.book.Serial,
                    status:response.data.book.Status
                });
            }).catch(error=>{
            })
    }
    /**
     * Make server request to edit a book by given id
     * @param {string} e
     * @returns {string} success message
     */

    handleSubmit = (event) => {
        event.preventDefault();


        const editbook = {
            "title":this.state.title,
            "author":this.state.author,
            "year":this.state.year,
            "serial_number":this.state.serial
        };
            const token = localStorage.getItem('access_token');
            const bookId = this.props.params.id;
            const edit_book_url = `http://127.0.0.1:5000/api/v2/books/${bookId}`
            const config ={ headers:{"Authorization":"Bearer " + token}}
            
            axios.put(edit_book_url, editbook, config)
            .then(response => {
                console.log(response);
                console.log(this.state);
                if(response.data.status === 200){
                    swal("Book details successfully updated");
                } else {
                    swal(response.data.message);
                }
            }).catch({});

        };

   render() {
    return (
        <div style={{ padding: '30px', color: '#337ab7' }}>
        <button className="btn btn-primary" onClick={browserHistory.goBack}>Go Back</button>
        <div className = "jumbotron" id="signupPage">
        <form onSubmit={this.handleSubmit}>
        <h2> Edit Book details </h2>
            <div className = "row">
                <div className ="col-xs-6">
                    <input
                    id="title"
                    className="form-control"
                    name="title"
                    type="text"
                    placeholder="Edit Book Title"
                    required={true}
                    value={this.state.title}
                    onChange={this.handleChange}
                    />
                </div>
            </div>
            <br/>
            <div className = "row">
                <div className ="col-xs-6">
                    <input
                    id="author"
                    className="form-control"
                    name="author"
                    type="text"
                    placeholder="Edit Book Author"
                    required={true}
                    value={this.state.author}
                    onChange={this.handleChange}
                    />
                </div>
            </div>
            <br/>
            <div className = "row">
                <div className ="col-xs-6">
                    <input
                        id="year"
                        className="form-control"
                        name="year"
                        type="number"
                        placeholder="Edit Publication Year"
                        required={true}
                        value={this.state.year}
                        onChange={this.handleChange}
                    />
                </div>
            </div>
            <br/>
            <div className = "row">
                <div className ="col-xs-6">
                    <input
                        id="serial"
                        className="form-control"
                        name="serial"
                        type="text"
                        placeholder="Edit Serial Number"
                        required={false}
                        value={this.state.serial}
                        onChange={this.handleChange}
                    />
                </div>
            </div>
            <br/>
            <button className='btn btn-primary' type="submit">Edit Book</button> 
        </form>
    </div>
    </div>

           );
        }
    }

export default EditBook;
