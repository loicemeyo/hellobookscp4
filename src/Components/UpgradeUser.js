import React from 'react';
import axios from 'axios';
import swal from 'sweetalert';
import { browserHistory } from 'react-router';

/**
 * This component enables a logged in admin to upgrade a normal user to admin status or downgrade them
 */

class UpgradeUser extends React.Component {
    state = {
        email: '',
    
    }
     /**
     * Allow the admin to view this function only when they are logged in.
     * Otherwise, redirect to login
     */
    componentDidMount (){
        const token = localStorage.getItem("access_token");
        if(!token){
            return browserHistory.push("/login")
        }
    }
    /**
     * This function sets the state to the new field value as entered by the user
     * @param {string} e
     * @returns {string} value
     */
    handleChange = (event) => {
        this.setState({
            [event.target.name]: event.target.value
        });
    }
    /**
     * Make server request to upgrade/downgrade a user
     * @param {int} bookId
     * @returns {object}book
     */
    handleSubmit = (event) => {
        event.preventDefault();

        const upgradinguser = {

            email: this.state.email,
    
        };

        const token = localStorage.getItem('access_token');
        const config ={ headers:{"Authorization":"Bearer " + token}}
    

        axios.put("http://127.0.0.1:5000//api/v2/auth/register", upgradinguser, config)
            .then(response => {
            console.log(response);
            console.log(this.state);
            browserHistory.push('/')
            if(response.data.status === 200){
                swal("You have successfully set this user's status");
            } else {
                swal(response.data.message);
            }

            }).catch(error => {
                console.log(error.response);
                if(error.response.status === 401){
                    swal (error.response.data.Message);
                }
                

            });


    };
    render() {
        return (
            <div style={{ padding: '30px', color: '#337ab7' }}>
            <button className="btn btn-primary" onClick={browserHistory.goBack}>Go Back</button>
            <div className="jumbotron" id="signupPage">
                <form onSubmit={this.handleSubmit}>
                    <h2> Change User Status</h2>
                    <div className="row">
                        <div className="col-xs-6">
                            <input
                                id="email"
                                name="email"
                                type="text"
                                placeholder="Enter Email"
                                required={true}
                                value={this.state.email}
                                onChange={this.handleChange}
                            />
                        </div>
                    </div>
                    <br />
                    <button className='btn btn-primary' type="submit">Change Status</button>
                </form>
            </div>
            </div>

        )
    }
}
export default UpgradeUser;