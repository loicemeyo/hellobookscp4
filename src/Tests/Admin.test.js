import React from "react";
import { shallow, mount } from "enzyme";
import sinon from "sinon";
import moxios from "moxios"
import shallowToJson, { mountToJson } from "enzyme-to-json";
import AddBook from "../Components/AddBook";
import Admin from "../Components/Admin";
import AllUsers from "../Components/AllUsers";
import UpgradeUser from "../Components/UpgradeUser";
import EditBook from "../Components/EditBook";

describe("Test Admin Component", () => {
  const wrapper = shallow(<Admin />);

  it("Admin component renders without crushing", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });

});
describe("Test AddBook Component", () => {
  beforeEach(() => {
    moxios.install();
  });
    
  afterEach(() => {
    moxios.uninstall();
  });
    
  it("Addbook component renders without crushing", () => {
    let wrapper = shallow(<AddBook />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });
  it("Calls handleSubmit() on AddBook form submission", () => {
    let wrapper = shallow(<AddBook />);
    let handleSubmit = sinon.spy();
    wrapper = mount(<AddBook onSubmit={handleSubmit} />);
    wrapper.find("form").simulate("submit");

    moxios.wait(() => { });
  });
  it("Calls handleChange() on book title input field", () => {
    let wrapper = shallow(<AddBook />);
    let handleChange = sinon.spy();
    wrapper = mount(<AddBook onChange={handleChange} />);
    wrapper.find("#title").simulate("change");
  })
});
describe("Test AllUsers Component", () => {
  beforeEach(() => {
    moxios.install();
  });

  afterEach(() => {
    moxios.uninstall();
  });
    
  it("AllUsers component renders without crushing", () => {
    const wrapper = shallow(<AllUsers />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });

});
describe("Test UpgradeUser Component", () => {
  let wrapper = shallow(<UpgradeUser />);

  it("UpgradeUser component renders without crushing", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });

});
describe("Test EditBook Component", () => {
  beforeEach(() => {
    moxios.install();
  });
    
  afterEach(() => {
    moxios.uninstall();
  });
  const params = {
    params: {
      id: 1
    }
  };
  it("EditBook component renders without crushing", () => {
    let wrapper = shallow(<EditBook params = {{ params }}/>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });
  //   it("Calls handleSubmit() on EditBook form submission", () => {
  //     let wrapper = shallow(<EditBook params = {{params}}/>);
  //     let handleSubmit = sinon.spy();
  //     wrapper = mount(<EditBook onSubmit={handleSubmit} />);
  //     wrapper.find("form").simulate("submit");

  //     moxios.wait(() => { });


  //   });

});