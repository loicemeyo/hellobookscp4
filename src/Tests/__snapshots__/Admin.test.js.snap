// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddBook Component Addbook component renders without crushing 1`] = `
<div
  style={
    Object {
      "color": "#337ab7",
      "padding": "30px",
    }
  }
>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Go Back
  </button>
  <div
    className="jumbotron"
    id="signupPage"
  >
    <form
      onSubmit={[Function]}
    >
      <h2>
         Add new book 
      </h2>
      <div
        className="row"
      >
        <div
          className="col-xs-6"
        >
          <input
            className="form-control"
            id="title"
            name="title"
            onChange={[Function]}
            placeholder="Enter Book Title"
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
      <br />
      <div
        className="row"
      >
        <div
          className="col-xs-6"
        >
          <input
            className="form-control"
            id="author"
            name="author"
            onChange={[Function]}
            placeholder="Enter Book Author"
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
      <br />
      <div
        className="row"
      >
        <div
          className="col-xs-6"
        >
          <input
            className="form-control"
            id="year"
            name="year"
            onChange={[Function]}
            placeholder="Enter Publication Year"
            required={true}
            type="number"
            value=""
          />
        </div>
      </div>
      <br />
      <div
        className="row"
      >
        <div
          className="col-xs-6"
        >
          <input
            className="form-control"
            id="serial"
            name="serial"
            onChange={[Function]}
            placeholder="Enter Serial Number"
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <br />
      <button
        className="btn btn-primary"
        type="submit"
      >
        Add Book
      </button>
    </form>
  </div>
</div>
`;

exports[`Test Admin Component Admin component renders without crushing 1`] = `
<div>
  <h2
    style={
      Object {
        "color": "#337ab7",
        "padding": "5px",
      }
    }
  >
    Admin Dashboard
  </h2>
  <div>
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/addbook"
    >
      <button
        className="btn btn-primary"
        id="button"
      >
        Add Book
      </button>
    </Link>
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/upgradeuser"
    >
      <button
        className="btn btn-primary"
        id="button"
      >
        Change user status
      </button>
    </Link>
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/users"
    >
      <button
        className="btn btn-primary"
        id="button"
      >
        View users
      </button>
    </Link>
  </div>
  <br />
  <h2
    style={
      Object {
        "color": "#337ab7",
        "padding": "5px",
        "textAlign": "center",
      }
    }
  >
    Available Books
  </h2>
  <div
    className="col=md-10"
  >
    <table
      className="table"
      id="table"
    >
      <thead>
        <tr>
          <th>
            Book
          </th>
          <th>
            Author
          </th>
          <th>
            Publication Year
          </th>
          <th>
            Edit Book
          </th>
          <th>
            Delete Book
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
  <div
    style={
      Object {
        "float": "right",
        "textDecoration": "none",
      }
    }
  >
    <t
      initialPage={1}
      items={Array []}
      labels={
        Object {
          "first": "First",
          "last": "Last",
          "next": "Next",
          "previous": "Previous",
        }
      }
      onChangePage={[Function]}
      pageSize={6}
    />
  </div>
</div>
`;

exports[`Test AllUsers Component AllUsers component renders without crushing 1`] = `
<div
  style={
    Object {
      "color": "#337ab7",
      "padding": "20px",
    }
  }
>
  <h2
    style={
      Object {
        "color": "#337ab7",
        "padding": "5px",
        "textAlign": "center",
      }
    }
  >
    HelloBooks Users
  </h2>
  <br />
  <div
    className="col=md-8"
  >
    <div
      className="card"
    >
      <table
        className="table"
        style={
          Object {
            "border": "1px solid grey",
          }
        }
      >
        <tr>
          <th>
            Name
          </th>
           
          <th>
            Email
          </th>
           
          <th>
            Admin
          </th>
        </tr>
      </table>
    </div>
  </div>
  <div
    style={
      Object {
        "float": "right",
        "textDecoration": "none",
      }
    }
  >
    <t
      initialPage={1}
      labels={
        Object {
          "first": "First",
          "last": "Last",
          "next": "Next",
          "previous": "Previous",
        }
      }
      onChangePage={[Function]}
      pageSize={6}
    />
  </div>
</div>
`;

exports[`Test EditBook Component EditBook component renders without crushing 1`] = `
<div
  style={
    Object {
      "color": "#337ab7",
      "padding": "30px",
    }
  }
>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Go Back
  </button>
  <div
    className="jumbotron"
    id="signupPage"
  >
    <form
      onSubmit={[Function]}
    >
      <h2>
         Edit Book details 
      </h2>
      <div
        className="row"
      >
        <div
          className="col-xs-6"
        >
          <input
            className="form-control"
            id="title"
            name="title"
            onChange={[Function]}
            placeholder="Edit Book Title"
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
      <br />
      <div
        className="row"
      >
        <div
          className="col-xs-6"
        >
          <input
            className="form-control"
            id="author"
            name="author"
            onChange={[Function]}
            placeholder="Edit Book Author"
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
      <br />
      <div
        className="row"
      >
        <div
          className="col-xs-6"
        >
          <input
            className="form-control"
            id="year"
            name="year"
            onChange={[Function]}
            placeholder="Edit Publication Year"
            required={true}
            type="number"
            value=""
          />
        </div>
      </div>
      <br />
      <div
        className="row"
      >
        <div
          className="col-xs-6"
        >
          <input
            className="form-control"
            id="serial"
            name="serial"
            onChange={[Function]}
            placeholder="Edit Serial Number"
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <br />
      <button
        className="btn btn-primary"
        type="submit"
      >
        Edit Book
      </button>
    </form>
  </div>
</div>
`;

exports[`Test UpgradeUser Component UpgradeUser component renders without crushing 1`] = `
<div
  style={
    Object {
      "color": "#337ab7",
      "padding": "30px",
    }
  }
>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Go Back
  </button>
  <div
    className="jumbotron"
    id="signupPage"
  >
    <form
      onSubmit={[Function]}
    >
      <h2>
         Change User Status
      </h2>
      <div
        className="row"
      >
        <div
          className="col-xs-6"
        >
          <input
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Enter Email"
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
      <br />
      <button
        className="btn btn-primary"
        type="submit"
      >
        Change Status
      </button>
    </form>
  </div>
</div>
`;
